### ISSUE TITLE
Can't override 'mountpoint' when mounting a filesystem

### ISSUE THREAD

**Issue by FransUrbo** [zfsonlinux/zfs#4553]
```
# zfs get mountpoint share/home/ftp/zol
NAME                PROPERTY    VALUE                    SOURCE
share/home/ftp/zol  mountpoint  /mnt/share/home/ftp/zol  inherited from share/home
# zfs mount -o mountpoint=/mnt/celia share/home/ftp/zol
filesystem 'share/home/ftp/zol' cannot be mounted due to invalid option 'mountpoint=/mnt/celia'.
Use the '-s' option to ignore the bad mount option.
cannot mount 'share/home/ftp/zol': Invalid argument
```

PS. Pool imported with <code>-N -R /mnt/share</code>. Setting <code>mountpoint</code> to the desired value and _then_ mounting it works.

This is <code>v0.6.5.6</code>.

**Comment by FransUrbo** [comment]
```
# zfs mount -s -o mountpoint=/mnt/share share/home/ftp
invalid option 's'
usage:
        mount
        mount [-vO] [-o opts] <-a | filesystem>

For the property list, run: zfs set|get

For the delegated permission list, run: zfs allow|unallow
```

**Comment by FransUrbo** [comment]
Looking at the manpage, I see:

```
   Temporary Mount Point Properties
       When a file system is mounted, either through mount(8) for legacy mounts or the zfs mount command for normal file systems, its  mount  options  are
       set according to its properties. The correlation between properties and mount options is as follows:

             PROPERTY                MOUNT OPTION
              devices                 devices/nodevices
              exec                    exec/noexec
              readonly                ro/rw
              setuid                  setuid/nosetuid
              xattr                   xattr/noxattr
              atime                   atime/noatime
              relatime                relatime/norelatime
              nbmand                  nbmand/nonbmand
```

So <code>mountpoint</code> isn't in there. But I think it's quite reasonable that it SHOULD be possible to use it.

**Comment by jf** [comment]
same for `0.6.5.9`, btw. And ditto - I'd like to have this.

**Comment by kpande** [comment]
`mount -t zfs -o zfsutil pool/filesystem /mnt/point`

**Comment by lucidBrot** [comment]
until this exists: It seems possible to use
```bash
mount -t zfs -o zfsutil rpool/dataset /mntpoint
```
to mount the `rpool/dataset` dataset at `/mntpoint`

**Comment by Haravikk** [comment]
Is it still not possible to change mountpoint using the `zfs mount` command directly? I just tried this today but instead of returning any kind of error like the OP mentioned, it just silently mounted the dataset using the mount point property of the dataset, rather than the option I set.

Personally I'd prefer the error if for some reason this functionality cannot be implemented, as an option supplied by the user should either work, or produce some kind of feedback; silence is dangerous. In my case I was trying to mount a backup dataset so I could run `zfs diff` on its snapshots (to find out what was eating up a load of space between snapshots) but instead of mounting at a harmless test location like I expected, it was mount *over* the live dataset the backups are created from! Fortunately it wasn't in active use, but I could see the current behaviour resulting in problems if it had been, as it would have suddenly become read-only.

I dunno if maybe I missed a memo somewhere about best practices for backups, but I like to preserve most of the properties in my backups, and the only property I changed is `readonly=on` (to prevent accidental changes to the contents of the backup), but this means that the mountpoint properties for my backup datasets are the same as the live ones, because I don't want to have to remember what they all are.

TL;DR
Please either use any mountpoint supplied via `-o`, or go back to returning an error to indicate that the option will not be used.

**Comment by almereyda** [comment]
It appears reopening this alongside #985 would seem feasible for the case described in the original posting.
